osier/models/capacity_expansion.py

```diff
@@ -83,21 +84,28 @@ def __init__(self,
                 prm=0.0,
                 penalty=LARGE_NUMBER,
                 power_units=u.MW, 
-                allow_blackout=True,
+                curtailment=True,
+                allow_blackout=False,
                 **kwargs):
         self.technology_list = deepcopy(technology_list)
         self.demand = demand
         self.prm = prm
-        self.max_demand = demand.max()
+        self.max_demand = float(demand.max())*power_units
         self.avg_lifetime = 25
         self.capacity_requirement = self.max_demand * (1+self.prm)
 
         self.objectives = objectives
         self.constraints = constraints
         self.penalty = penalty
         self.power_units = power_units
+        self.curtailment = curtailment
         self.allow_blackout = allow_blackout
 
+        if isinstance(demand, unyt_array):
+            self.power_units = demand.units
+        else:
+            self.power_units = power_units
+
```

Comment:
Is line 100 still needed because you have lines 104-108?

Reply:
Yes because it’s not guaranteed that demand will be passed as a unyt array. If it is, then these lines override default power units.

Reply:
But doesn't the `else` part of that loop take care of that for you?

Reply:
I suppose that's true, but this way, it's always defined. If it makes you and @yardasol happy, I will get rid of line 100.